vimPlugins.onebuddy: init at 2021-04-01

jfchevrette opened the pull request:
###### Motivation for this change

Add https://github.com/Th3Whit3Wolf/onebuddy

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

figsoda commented on pkgs/misc/vim-plugins/generated.nix:
```diff
@@ -7163,7 +7175,7 @@ final: prev:
       repo = "vim-colorschemes";
       rev = "fd8f122cef604330c96a6a6e434682dbdfb878c9";
-      sha256 = "1cg8q7w0vgl73aw1b9zz0zh5vw5d2pm8pm54fhfzva4azg56f416";
+      sha256 = "0kpsf6j20fgblc8vhqn7ymr52v2d1h52vc7rbxmxfwdm80nvv3g5";
```
This causes a hash mismatch, also see as https://github.com/NixOS/nixpkgs/pull/145640#issuecomment-971126083
somehow this is the exact same hash, is there something wrong with the update script?

cc @dotlambda @lovesegfault @chuahou from the previous conversations

lovesegfault replied:
:ghost:

chuahou replied:
This is really strange... @dotlambda and I also had very different results running update.py minutes apart from each other. It shouldn't be different nixpkgs versions since I'm assuming we're all running it on master, and `update-shell.nix` refers to the current nixpkgs ref's packages, so perhaps it is something to do with the GitHub access token used? (very wild guess)

chuahou replied:
Something funky is going on with how vim-colorschemes is being fetched. I get [the same hash mismatch](https://github.com/NixOS/nixpkgs/pull/146342#issuecomment-971335015) when building it from this commit myself, but ofborg [had no such trouble](https://gist.github.com/GrahamcOfBorg/3e971c938a1ddbfc15574d64226f7031).

chuahou replied:
Running update.py again on top of this commit reverts the sha256 change, as in line 7178 of https://github.com/ch-fork/nixpkgs/commit/a7b84f9843d2f3dd8489ec11ca1c6b37f8991adc.

jfchevrette replied:
The word `Update` is capitalized because I followed these instructions https://nixos.wiki/wiki/Vim#Adding_new_plugins
And indeed initially I ran update.py with `--no-commit` because I wanted to inspect the changes and do it manually to learn how it works

chuahou commented:
@jfchevrette Could you please let us know exactly what command(s) you ran to generate these commits?

jfchevrette committed:
vimPlugins: Update

jfchevrette committed:
vimPlugins.onebuddy: init at 2021-04-01

jfchevrette commented:
@chuahou I used the following from master. My token pas `public_repo` access only
```
GITHUB_API_TOKEN=ghp_foo ./pkgs/misc/vim-plugins/update.py
```

I just rebased on top of master and updated the vimPlugins update commit by re-running the above and then re-applied the new `onebuddy` package on top. Does this look better now? Hopefully I did not mess something up I'm fairly new to this :)

chuahou commented:
Interesting to see that rerunning update.py for you still results in the weird sha256, that's weird. Let's see if ofborg-eval does anything different this time around?

dotlambda commented:
Maybe this is related to case-insensitive filesystems on macOS?

figsoda commented:
> Maybe this is related to case-insensitive filesystems on macOS?

that's probably it, both `darkblue.vim` and `darkBlue.vim` exist under `colors`

figsoda commented:
relevant issue upstream: https://github.com/flazz/vim-colorschemes/issues/186

jfchevrette commented:
I did run this from an aarch64-darwin machine, and both my / and /nix filesystems are APFS and case-insensitive (which is the default)

chuahou commented:
Ah, that makes sense. I wonder why ofborg-eval had no trouble building it for `x86_64-linux`, though?

teto commented:
feel free to reopen if you rebase/update the MR
